### ISSUE TITLE
Ability to Export classes from one schema, but run in another

### ISSUE THREAD

**Issue by cowwoc** [querydsl/querydsl#1048]
I am familiar with the ability to override the schema for QClasses at runtime using https://github.com/querydsl/querydsl/issues/611 but what I am looking for is the ability to do the following:
1. Use Flyway to export my database migrations to a schema called "querydsl". This migration only creates tables. All data is excluded.
2. Use QueryDSL to generate QClasses by reading tables from schema "querydsl" but have the QClasses actually run in a secondary schema "public" at runtime.

Use-case: I want to reuse business logic for populating tables. Instead of populating tables once using SQL migrations, then implementing the same logic in Java classes for populating tables at runtime, I'd like to populate tables exclusively from Java migration classes. See http://flywaydb.org/documentation/migration/java.html

This way, whether I am adding the first user (administrator) at build-time, or a dynamic user at runtime, the same Java code will run.
1. My plan is to provide SQL migrations named V1, V2, V3 that would create the tables but do nothing else. I would then provide Java migrations V1_1, V2_1, V3_1 that would populate the tables.
2. At build-time, Flyway creates a schema called "querydsl" and only runs the SQL migrations. It then runs QueryDSL Export to generate QClasses from this schema.
3. At runtime, Flyway runs both the SQL and Java migrations and operations against a schema called "public".

So to recap: I want QueryDSL Export to read the database schema from "querydsl" but have the resulting QClasses operate on the "public" schema.

I hope you understand what I mean. If not, please ask for clarification and I will try to provide more information.

Right now there is only one configuration property (`<schemaPattern>`) which denotes both the input and output schema.

I am also open to suggestions if you can think of a simpler solution for implementing this use-case.
